src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -171,7 +171,7 @@ class Plugin(RepositoryPlugin[TDRSourceSpec, TDRSourceRef]):
     @classmethod
     def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
         return cls(sources=frozenset(
-            TDRSourceSpec.parse(spec).effective
+            TDRSourceSpec.parse(spec)
```

Comment:
@hannes-ucsc 
removing .effective here causes failures during the `.list_bundle` operation, we cannot assert that the effective source from the notification matches the sources from the config class because its possible that the partition is set to None.

Reply:
https://github.com/DataBiosphere/azul/blob/develop/src/azul/plugins/repository/tdr/__init__.py#L226

Should this assert be changed to inspect the SourceRef.id rather than the spec?